benchmarks/src/commonMain/kotlin/SegmentPoolBenchmark.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.io.benchmarks
+
+import kotlinx.io.*
+import kotlinx.benchmark.*
+
+@State(Scope.Benchmark)
+open class SegmentPoolBenchmark {
+    private val buffer = Buffer()
+
+    @Benchmark
+    fun acquireReleaseCycle() {
```

Comment:
If this test is run in parallel with the others, will they affect each other (for example, if common static fields are used)?

Reply:
This one will not be executed concurrently unless the user will explicitly specify the opposite. And in concurrent case it may fail due to lack of synchronization.

Not sure if there is a way to confine benchmark's state to a single thread using `kotlinx.benchmark`'s annotations (like it could be done using JMH's `@Scope(Thread)`.